core/src/test/java/io/grpc/internal/ReflectionLongAdderCounterTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.internal;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ReflectionLongAdderCounterTest {
+  private ReflectionLongAdderCounter counter;
+
+  @Before
+  public void setUp() {
+    counter = new ReflectionLongAdderCounter();
```

Comment:
init it inline above?

Reply:
This `@Before` creates a new counter at each test iteration

Reply:
I think there is a new instance created each test too

Reply:
I didn't know junit does that, TIL

Reply:
That's the number one benefit to JUnit 4 to me. JUnit 3 didn't behave that way.